[#152632922] Remove file extensions from product page links

dcarley opened the pull request:
## What

Links to the product pages shouldn't have a file extension because
Middleman renders them without and that's how they are referenced in the
product pages themselves.

Most of these are in the header, but there were also some references to the
support page within the custom domain documentation.

These also caused the requests to redirect off of `service.gov.uk` and go to
`cloudapps.digital` because of a bug that we're fixing in a separate story.

## How to review

Describe the steps required to test the changes.

1. Preview the content locally ([see README](https://github.com/alphagov/paas-tech-docs#preview)) and check that it renders as expected.
1. Follow all of the links in the top-right header and check that they work.
1. Follow all of the support links in [/#using-a-custom-domain](http://localhost:4567/#using-a-custom-domain) and check that they work.

## Who can review

Anyone.

alext commented:
Changes look good, I'm not sure the commit message is correct though. The product pages are no longer rendered by middleman - they're now a Sinatra app. I believe it was this change that changed the URLs.

dcarley commented:
> Changes look good, I'm not sure the commit message is correct though. The product pages are no longer rendered by middleman - they're now a Sinatra app. I believe it was this change that changed the URLs.

ACK, removed the referenced and rebased.
